src/Eloquent/Mixins/BuilderReturning.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tpetry\PostgresqlEnhanced\Eloquent\Mixins;
+
+use Closure;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Tpetry\PostgresqlEnhanced\Support\Helpers\ModelHelper;
+
+/** @mixin \Illuminate\Database\Eloquent\Builder */
+class BuilderReturning
+{
+    public function deleteReturning(): Closure
+    {
+        return function (array $returning = ['*']): Collection {
+            /** @var \Illuminate\Database\Eloquent\Builder $this */
+            $time = $this->getModel()->freshTimestampString();
+            $modelHelper = app(ModelHelper::class);
+
+            if ($deletedAtColumn = $modelHelper->getDeletedAtColumn($this->getModel())) {
+                return $this->updateReturning([$deletedAtColumn => $time], $returning);
+            }
+
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->deleteReturning(null, $returning)->all()
+            )->each(fn (Model $model) => $model->exists = false);
+        };
+    }
```

Comment:
I think we need a forceDeleteReturning

```suggestion
    public function deleteReturning(): Closure
    {
        return function (array $returning = ['*']): Collection {
            /** @var \Illuminate\Database\Eloquent\Builder $this */
            $time = $this->getModel()->freshTimestampString();
            $modelHelper = app(ModelHelper::class);

            if ($deletedAtColumn = $modelHelper->getDeletedAtColumn($this->getModel())) {
                return $this->updateReturning([$deletedAtColumn => $time], $returning);
            }

            return $this->forceDeleteReturning($returning);
        };
    }
    
    public function forceDeleteReturning(): Closure
    {
        return function (array $returning = ['*']): Collection {
            /** @var \Illuminate\Database\Eloquent\Builder $this */
            return $this->hydrate(
                $this->applyScopes()->getQuery()->deleteReturning(null, $returning)->all()
            )->each(fn (Model $model) => $model->exists = false);
        };
    }
```